homeassistant/components/bosch_shc/sensor.py

```diff
@@ -0,0 +1,401 @@
+            PowerSensor(
+                device=sensor,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+            )
+        )
+        entities.append(
+            EnergySensor(
+                device=sensor,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+            )
+        )
+
+    for sensor in session.device_helper.smart_plugs_compact:
+        entities.append(
+            PowerSensor(
+                device=sensor,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+            )
+        )
+        entities.append(
+            EnergySensor(
+                device=sensor,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+            )
+        )
+
+    for sensor in (
+        session.device_helper.smoke_detectors
+        + session.device_helper.shutter_contacts
+        + session.device_helper.universal_switches
+        + session.device_helper.thermostats
+        + session.device_helper.wallthermostats
+        + session.device_helper.twinguards
+    ):
+        if sensor.supports_batterylevel:
+            entities.append(
+                BatterySensor(
+                    device=sensor,
+                    parent_id=session.information.unique_id,
+                    entry_id=config_entry.entry_id,
+                )
+            )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class TemperatureSensor(SHCEntity):
+    """Representation of a SHC temperature reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_temperature"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Temperature"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.temperature
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return TEMP_CELSIUS
+
+
+class HumiditySensor(SHCEntity):
+    """Representation of a SHC humidity reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_humidity"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Humidity"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.humidity
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return PERCENTAGE
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return "mdi:water-percent"
```

Comment:
We normally don't set a custom icon when setting device class. It will already have a specific icon in the frontend.